lib/contexts/tokenPrice.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react';
+
+import useFetch from '../hooks/useFetch';
+import isBrowser from '../isBrowser';
+
+const TOKEN_PRICE_LS_KEY = 'token_price_cache';
+const BinanceRatesUrl = 'https://api.binance.com/api/v3/klines?symbol=ONEUSDT&interval=1d&limit=1000';
+const DayMs = 24 * 60 * 60 * 1000;
+
+interface TokenPrice {
+  price: number;
+  timestamp: number;
+}
+
+interface ProviderProps {
+  children: React.ReactNode;
+}
+
+type TimestampValue = number | string | null
+
+interface TAddressFormatContext {
+  getPriceByTimestamp: (timestamp: TimestampValue) => number | null;
+}
+
+const getValuesFromLS = () => {
+  try {
+    const data = localStorage.getItem(TOKEN_PRICE_LS_KEY);
+    if (data) {
+      return JSON.parse(data) as Array<TokenPrice>;
+    }
+  } catch (e) {}
+  return [];
+};
+
+const saveValuesToLs = (values: Array<TokenPrice>) => {
+  localStorage.setItem(TOKEN_PRICE_LS_KEY, JSON.stringify(values));
+};
+
+export const TokenPriceContext = React.createContext<TAddressFormatContext>({
+  getPriceByTimestamp: () => {
+    return null;
+  },
+});
+
+export function TokenPriceProvider({ children }: ProviderProps) {
+  const [ values, setValues ] = useState<Array<TokenPrice>>([]);
+  const fetch = useFetch();
+
+  const browser = isBrowser();
+
+  useEffect(() => {
+    const updateCache = async() => {
+      try {
+        const valuesFromLS = getValuesFromLS();
+        const lastValueFromLS = valuesFromLS[valuesFromLS.length - 1];
+        // Fetch token prices history from Binance
+        if (
+          !lastValueFromLS ||
+          (Date.now() - lastValueFromLS.timestamp > DayMs)
+        ) {
+          const result = await fetch<Array<Array<number>>, unknown>(BinanceRatesUrl);
+          if (Array.isArray(result)) {
+            const newValues: Array<TokenPrice> = result.map(([ timestamp, price ]) => {
+              return {
+                timestamp,
+                price,
+              };
+            });
+            setValues(newValues);
+            saveValuesToLs(newValues);
+          }
+        } else {
+          // Restore values from LS
+          setValues(valuesFromLS);
+        }
+      } catch (e) {}
```

Comment:
Empty catch with no error state management. I would suggest adding some logic that sets values to default. It will be clear to the developer what should happen.